modules/c8000v/variables.tf

```diff
@@ -0,0 +1,146 @@
+variable "metro_code" {
+  description = "Device location metro code"
+  type        = string
+  validation {
+    condition     = can(regex("^[A-Z]{2}$", var.metro_code))
+    error_message = "Valid metro code consists of two capital letters, i.e. SV, DC."
+  }
+}
+
+variable "project_id" {
+  description = "project_id"
+  type        = string
+}
+
+variable "account_number" {
+  description = "Billing account number for a device"
+  type        = string
+  default     = 0
+}
+
+variable "platform" {
+  description = "Device platform flavor that determines number of CPU cores and memory"
+  type        = string
+  validation {
+    condition     = can(regex("^(small|medium|large)$", var.platform))
+    error_message = "One of following platform flavors are supported: small, medium, large."
+  }
+}
+
+variable "software_package" {
+  description = "Device software package"
+  type        = string
+  validation {
+    condition     = can(regex("^(network-advantage|network-essentials|network-premier)$", var.software_package))
+    error_message = "One of following software packages are supported: STD."
+  }
+}
+
+variable "name" {
+  description = "Device name"
+  type        = string
+  validation {
+    condition     = length(var.name) >= 2 && length(var.name) <= 50
+    error_message = "Device name should consist of 2 to 50 characters."
```

Comment:
I think the default maximum value is 30 characters.

Reply:
I tested , its min 3 to 50 chars